Salwador/Views/ImageGenerationView.swift

```diff
@@ -88,29 +85,29 @@ private extension ContentView {
 
     var imageView: some View {
         VStack {
-            if hasError {
+            if viewModel.hasError {
                 HStack {
                     Image(systemName: "bell")
                         .foregroundColor(.red)
                         .scaledToFit()
                         .frame(width: 48, height: 48)
-                    Text(errorMsg)
+                    Text(viewModel.errorMsg)
                         .foregroundColor(Color("TextColor"))
                 }
-            } else if let image {
-                if isLoading {
+            } else if let identifier = viewModel.image {
+                if viewModel.isLoading {
```

Comment:
praise: 👍 for starting to move to ViewMode, continue. Suggestion: not use primitive types of `Bool` for this. Make your own type. A Enum would be a good fit for such a thing. 

Here is a follow up on this https://betterprogramming.pub/manage-view-state-with-enums-9034e461c5fc